### PR TITLE
Bump peewee to 3.17.*

### PR DESCRIPTION
**Pull request by sobolevn**
Closes https://github.com/python/typeshed/pull/10884

### CODE REVIEW

File: stubs/peewee/peewee.pyi [hunk 1]
```diff
@@ -116,9 +117,10 @@ class Node:
     def unwrap(self): ...
 
 class ColumnFactory:
-    node: Incomplete
-    def __init__(self, node) -> None: ...
-    def __getattr__(self, attr: str): ...
+    node: Node
+    def __init__(self, node: Node) -> None: ...
+    def __getattr__(self, attr: str) -> Any: ...
+    def __getitem__(self, attr: str) -> Any: ...
```
Comment (srittau) [inline stubs/peewee/peewee.pyi:123]: Shouldn't this be:

```suggestion
    def __getattr__(self, attr: str) -> Column: ...
    def __getitem__(self, attr: str) -> Column: ...
```

File: stubs/peewee/peewee.pyi [hunk 2]
```diff
@@ -880,8 +882,12 @@ class SqliteDatabase(Database):
...
     def register_collation(self, fn, name: Incomplete | None = ...): ...
     def collation(self, name: Incomplete | None = ...): ...
-    def register_function(self, fn, name: Incomplete | None = ..., num_params: int = ...) -> None: ...
-    def func(self, name: Incomplete | None = ..., num_params: int = ...): ...
+    def register_function(
+        self, fn, name: Incomplete | None = ..., num_params: int = ..., deterministic: bool | None = None
+    ) -> None: ...
+    def func(
+        self, name: Incomplete | None = ..., num_params: int = ..., deterministic: bool | None = None
+    ) -> Callable[[_F], _F]: ...
```
Comment (srittau) [inline stubs/peewee/peewee.pyi:890]:
```suggestion
    def register_function(
        self, fn, name: str | None = None, num_params: int = -1, deterministic: bool | None = None
    ) -> None: ...
    def func(
        self, name: str | None = None, num_params: int = -1, deterministic: bool | None = None
    ) -> Callable[[_F], _F]: ...
```

**Review by srittau — COMMENTED** [review]
Thanks, a few suggestions below.

File: stubs/peewee/peewee.pyi (hunk 2 above)
Comment (sobolevn) [reply stubs/peewee/peewee.pyi:890]: `name` is passed to `create_function` in `sqlite` which has `str` annotation 👍

**Review by srittau — APPROVED** [review]

**Pull request merged**